src/VisualStudio/Core/Def/CallHierarchy/CallHierarchyCommandHandler.cs

```diff
@@ -59,22 +60,25 @@ public bool ExecuteCommand(ViewCallHierarchyCommandArgs args, CommandExecutionCo
                     return true;
                 }
 
-                var semanticModel = document.GetSemanticModelAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+                var semanticModel = _threadingContext.JoinableTaskFactory.Run(() => document.GetSemanticModelAsync(cancellationToken));
```

Comment:
Should we just surround the entire method more or less with a single JoinableTaskFactory.Run() rather than doing it piecemeal for each call like this?

Reply:
@akhera99  feel free to reach out and we can discuss the approach here :)